File: app/src/main/java/com/arcgistest/di/components/ApplicationComponent.java
package com.arcgistest.di.components;

import android.content.Context;

import com.arcgistest.di.modules.ApplicationDIModule;
import com.arcgistest.di.modules.InteractorModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {
        ApplicationDIModule.class,
        InteractorModule.class
})
public interface ApplicationComponent {

    Context context();

    void inject(ComponentHolder holder);
}

File: app/src/main/java/com/arcgistest/di/components/ApplicationComponent.kt
package com.arcgistest.di.components

import android.content.Context

import com.arcgistest.di.modules.ApplicationDIModule

import javax.inject.Singleton

import dagger.Component

@Singleton
@Component(modules = arrayOf(ApplicationDIModule::class))
interface ApplicationComponent {

    fun context(): Context

    fun inject(holder: ComponentHolder)
}
